scenes/game/game.gd

```diff
@@ -1800,24 +1801,43 @@ func _on_reshuffle_deck_mulligan(_event):
 	#printlog("UI: TODO: In place reshuffle deck. No cards actually move though.")
 	pass
 
+func reset_revealed_cards():
+	var current_children = $AllCards/OpponentRevealed.get_children()
+	for i in range(len(current_children)-1, -1, -1):
+		var card = current_children[i]
+		card.get_parent().remove_child(card)
+		card.queue_free()
+
+func add_revealed_card(card_game_object):
+	var card_db = game_wrapper.get_card_database()
+	var logic_card : GameCard = card_db.get_card(card_game_object.card_id)
+	var copy_card = create_card(card_game_object.card_id + RevealCopyIdRangestart, logic_card.definition, card_game_object.card_image, card_game_object.cardback_image, $AllCards/OpponentRevealed, 0, true)
+	copy_card.set_card_and_focus(OffScreen, 0, CardBase.ReferenceCardScale)
+	copy_card.resting_scale = CardBase.ReferenceCardScale
+	copy_card.change_state(CardBase.CardState.CardState_Offscreen)
+	copy_card.flip_card_to_front(true)
+
+func _on_reveal_card_from_hand(event):
+	var player = event['event_player']
+	var card_id = event['number']
+	spawn_damage_popup("Card Revealed!", player)
+	if player == Enums.PlayerId.PlayerId_Opponent:
+		reset_revealed_cards()
```

Comment:
is there any case where someone might reveal several cards back-to-back? maybe would be nice to only reset on the first reveal of a new turn

Reply:
This is one of the UI parts that needs some additional love which I won't address in this change.